### PR TITLE
Clown Bomb Clowns and Station Clown share a faction, stopping attacks

### PR DESCRIPTION
**Pull request by BongoDongo2000**
The clown bomb will spawn a lovely whopping 50 clowns for you to play with. Unfortunately these clowns don't seem to recognize the station clown as a friend! The unknowing station clown tries to blend in with the crowd, but they single them out and beat them with rubber chickens. A single tear rolls down the station clown's cheek as the others wander away honking from the clown's broken frame.

Now, no more!

## About The Pull Request

Added the "clowns" faction to the clown mobs spawned by the clown specific traitor item the clown bomb.
Added the "clowns" faction to the station clown.

## Why It's Good For The Game

As things stand, the clowns spawned by the clown bomb are not hostile until attacked. At which point they retaliate by attacking anyone around them. 

Now, they won't attack those in the "clowns" faction. This lets the traitor clown have some more fun and hide in the crowd of clowns. 

## Changelog

:cl:

feature: Station Clowns will no longer be targets of aggression for Clown Bomb Clowns as they now share a faction. This may or may not lead to an uptick in clown bombing, both the honking and depressurizing kind.
wip: A new /_DEFINES/mobfactions.dm to define mob factions. Currently only has FACTION_CLOWN, will be expanded.

/:cl:

### CODE REVIEW

**Comment by Livrah** [comment]
What about clown ops? They have access to clown bomb too.

File: code/modules/jobs/job_types/clown.dm [hunk 1]
```diff
@@ -2,7 +2,7 @@
 	title = JOB_CLOWN
 	description = "Entertain the crew, make bad jokes, go on a holy quest to find bananium, HONK!"
 	department_head = list(JOB_HEAD_OF_PERSONNEL)
-	faction = FACTION_STATION
+	faction = list(FACTION_STATION,"clowns")
```
Comment (Fikou) [inline code/modules/jobs/job_types/clown.dm:5]: create a clown faction define

Comment (BongoDongo2000) [reply code/modules/jobs/job_types/clown.dm:5]: The issue with that is where to place it. There is no /code/_DEFINES/factions.dm. For instance the faction defines for antags is in

https://github.com/tgstation/tgstation/blob/5b6fa14a9f0d9c9c8b783a4453d602420482e9c2/code/__DEFINES/antagonists.dm

**Comment by BongoDongo2000** [comment]
> What about clown ops? They have access to clown bomb too.

I'm unsure how the selection on antagonists work exactly and wasn't sure how to work it in. I will likely at a later date go back and do so, along with adding the clown faction to the rest of the clown mobs.

**Comment by Shadowflame909** [comment]
if clown ops can't be added. Maybe you can add the faction to the special clown-op hardsuit. At least I think that's how the curator/traitor bundle space-carp suit has carp faction immunity

**Comment by Serxule** [comment]
isnt it specifically listed in the description of the clown bomb that itll attack you?

**Comment by BongoDongo2000** [comment]
> if clown ops can't be added. Maybe you can add the faction to the special clown-op hardsuit. At least I think that's how the curator/traitor bundle space-carp suit has carp faction immunity

It should be able to be added to the clown ops, again I'm just not sure on how the antagonist selection code functions there exactly for clown ops. I can revisit that and do so at another date.



> isnt it specifically listed in the description of the clown bomb(s) that itll attack you?

honk
HONK HONK
**HONK** **HONK** **HONK** **HONK** **HONK** **HONK** **HONK**

No. It just says "HONK." for the description.
https://github.com/tgstation/tgstation/blob/8f3121454b65e79994d8920b83784fe1950d5fbb/code/game/machinery/syndicatebomb.dm
Lines 251-256

The description in the syndicate uplink doesn't mention that either.
https://github.com/tgstation/tgstation/blob/639095036a0a0dbdf46a09ea2e477f2836c892fe/code/modules/uplink/uplink_items/job.dm
Lines 232-242

File: code/modules/jobs/job_types/clown.dm (hunk 1 above)
Comment (SuperNovaa41) [reply code/modules/jobs/job_types/clown.dm:5]: Make one then?

Comment (BongoDongo2000) [reply code/modules/jobs/job_types/clown.dm:5]: This is babies first PR. I mean I could but then i'd be changing even more stuff and possibly screwing stuff up. The heretic faction define lives in the antagonists.dm, right next to the other heretic defines, same for other faction defines which makes sense.

**Comment by BongoDongo2000** [comment]
After talking with Fikou, I've removed adding the "clowns" faction directly in the faction lists and instead moved it to a "post_equip" action.

**Comment by BongoDongo2000** [comment]
Ok i was retarded but it functions now

File: code/modules/jobs/job_types/clown.dm (hunk 1 above)
Comment (Kylerace) [reply code/modules/jobs/job_types/clown.dm]: make one please

Comment (BongoDongo2000) [reply code/modules/jobs/job_types/clown.dm]: Its done

**Comment by BongoDongo2000** [comment]
Added a new /_defines/mobfactions.dm to include a FACTION_CLOWN #define. 
Its empty for now, but I'll fill it out and change the faction variable for mobs on my next pr/commit.

File: code/modules/jobs/job_types/clown.dm [hunk 2]
```diff
@@ -97,3 +97,4 @@
 		M.mutadone_proof = TRUE
 	var/datum/atom_hud/fan = GLOB.huds[DATA_HUD_FAN]
 	fan.add_hud_to(H)
+	H.faction += FACTION_CLOWN
```
Comment (Fikou) [inline code/modules/jobs/job_types/clown.dm:100]:
```suggestion
	H.faction |= FACTION_CLOWN
```

Comment (BongoDongo2000) [reply code/modules/jobs/job_types/clown.dm]: Done.

**Pull request merged**